cmd/kubestellar-list-mid-dynamic/main.go

```diff
@@ -0,0 +1,180 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+// Import of k8s.io/client-go/plugin/pkg/client/auth ensures
+// that all in-tree Kubernetes client auth plugins
+// (e.g. Azure, GCP, OIDC, etc.)  are available.
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/pflag"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	machschema "k8s.io/apimachinery/pkg/runtime/schema"
+	k8sdynamic "k8s.io/client-go/dynamic"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	upstreamcache "k8s.io/client-go/tools/cache"
+	_ "k8s.io/component-base/metrics/prometheus/clientgo"
+	"k8s.io/klog/v2"
+
+	kcpdynamic "github.com/kcp-dev/client-go/dynamic"
+	kcpscopedclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+
+	clientopts "github.com/kubestellar/kubestellar/pkg/client-options"
+	"github.com/kubestellar/kubestellar/pkg/mailboxwatch"
+)
+
+const mainName = "kubestellar-list-mid-dynamic"
+
+var watch bool
+var outputJson bool
+
+func main() {
+	fs := pflag.NewFlagSet(mainName, pflag.ExitOnError)
+	gvr := machschema.GroupVersionResource{Version: "v1"}
+	var kind string
+	klog.InitFlags(flag.CommandLine)
+	fs.AddGoFlagSet(flag.CommandLine)
+
+	fs.StringVar(&gvr.Group, "api-group", gvr.Group, "API group of objects to watch (default is Kubernetes core group)")
+	fs.StringVar(&gvr.Version, "api-version", gvr.Version, "API version (just version, no group) of objects to watch")
+	fs.StringVar(&gvr.Resource, "api-resource", gvr.Resource, "API resource (lowercase plural) of objects to watch (defaults to lowercase(kind)+'s')")
+	fs.StringVar(&kind, "api-kind", kind, "kind of objects to watch")
+	fs.BoolVar(&watch, "watch", watch, "indicates whether to inform rather than just list")
+	fs.BoolVar(&outputJson, "json", outputJson, "indicates whether to output as lines of JSON rather than YAML")
+
+	parentClientOpts := clientopts.NewClientOpts("parent", "access to the parent of mailbox workspaces")
+	parentClientOpts.SetDefaultCurrentContext("root")
+	parentClientOpts.AddFlags(fs)
+
+	allClientOpts := clientopts.NewClientOpts("all", "access to the chosen objects in all clusters")
+	allClientOpts.SetDefaultCurrentContext("system:admin")
+	allClientOpts.AddFlags(fs)
+
+	fs.Parse(os.Args[1:])
+
+	ctx := context.Background()
+	logger := klog.Background()
+	ctx = klog.NewContext(ctx, logger)
+
+	if len(kind) == 0 {
+		logger.Error(nil, "The --kind must not be the empty string")
+	}
+	if len(gvr.Resource) == 0 {
+		gvr.Resource = strings.ToLower(kind) + "s"
+	}
+
+	parentClientConfig, err := parentClientOpts.ToRESTConfig()
+	if err != nil {
+		logger.Error(err, "failed to make ESPW client config")
```

Comment:
Maybe change from 'ESPW' to 'parent' here as well?

Reply:
Thanks, fixed.